### ISSUE TITLE
Spurious Sketcher toolbar appears above the Ribbon UI 

### ISSUE THREAD

**Issue by raindropsfromsky** [APEbbers/FreeCAD-Ribbon#135]
This happens occasionally. (Not reproducible.)
So far, this has happened three times.

How this happened:
1. I had added a body and then a sketch in the **Part Design** WB.
2. FC switched to Sketcher 
3. I added an arc. 

This is when the Sketcher toolbar appeared above the Ribbon UI.
![image](https://github.com/user-attachments/assets/879ad5b8-a489-4846-a336-f28e1d26f823)

BTW, it disappeared on its own.
I did not have to do anything.

**Comment by APEbbers** [comment]
I'll look into it. There is a function that hides all toolbars when you activate a tab. Maybe that needs to be more rigid somehow
Only toolbars in the statusbar are currently allowed.

**Comment by raindropsfromsky** [comment]
I found a pattern!
 
I have set FC to start with the **Part design** WB.

My sequence of operation is as follows:
1. Press **CTRL+N** to launch a new file.
2. In the **Tasks** panel, click on **Create body**
3. In the **Tasks** panel, click on **Create sketch**
4. Click on any plane (e.g. XY plane).
    At this point, all toolbars of Sketcher wB appear above the Ribbon UI.
5. If I close the Sketch in the **Tasks** panel, the Sketcher toolbars disappear.
6. If I double-click on the **Sketch** node in the **Model** panel, the Sketcher toolbars reappear.
7. If I close the Sketch in the **Tasks** panel, the Sketcher toolbars disappear.

To sum up, FC shows the Sketcher toolbars whenever the Sketcher WB is launched from the tree or Tasks panel.
But, if I select the Sketcher WB from the Ribbon UI, the spurious toolbars do NOT appear!

**Comment by APEbbers** [comment]
That's strange. When I do the same, they don't appear. They stay away.

If you try to disable them one by one when you at step for, close and reopen the sketch. Do they still appear?

**Comment by APEbbers** [comment]
If you go to the parameter editor under tools, what do you see under "mainwindow" -> toolbar? at my laptop, they are set to False.
![image](https://github.com/user-attachments/assets/02aa1c71-fe50-462a-9da1-ae7979156d6e)

**Comment by raindropsfromsky** [comment]
> If you try to disable them one by one when you at step for, close and reopen the sketch. Do they still appear?

No. When I right-clicked on one of the Sketcher toolbars, it showed me a long list of toolbars, with checkboxes.
The toolbars which appeared above the Ribbon UI had ticks in their checkboxes.
I unselected all the checkboxes one by one till all Sketcher toolbars disappeared.
Now if I repeat the steps described above, the toolbars do not appear anymore.

**Comment by APEbbers** [comment]
Nice!, I just did a test: I set one toolbar to True in the parameters. Then it did appear like with you. 

Now I know how to solve it thanks for the help!

**Comment by raindropsfromsky** [comment]
> If you go to the parameter editor under tools, what do you see under "mainwindow" -> toolbar? at my laptop, they are set to False.

After I closed the Sketcher toolbars manually, all the parameters had "false" values.
So combining our observations, it means that FC was somehow setting the initial values of only Sketcher parameters to "true".

So is the solution to initialize all WBs to false through a script?

**Comment by APEbbers** [comment]
Yes, indeed. there is already a function that hides the toolbars.
i can extend it to also set the parameters to false. that way, they cannot appear without being enabled by the user.

**Comment by MarcelP102** [comment]
Hi @APEbbers with latetst dev the draft toolbar appears above the ribbon:
![afbeelding](https://github.com/user-attachments/assets/4243d25b-390d-4ccc-a14a-40787bade659)

Some time ago I also had the issue with sketcher and I fixed it with the parameter editor

**Comment by APEbbers** [comment]
Hi, 

Persistent toolbars are now removed on startup. This should solve the issue with the draft toolbar.
FYI: The ribbon does allow toolbars inside the statusbar. for if you want to place a view toolbar there for example.

Some workbenches automatically load toolbars in other workbenches. If that is the case, they often have a setting to switch this of. DynamicData is an example.

**Comment by raindropsfromsky** [comment]
Sketcher toolbar still appears above the UI.

OS: Windows 11 build 22631
Architecture: x86_64
Version: 1.1.0dev.39708 (Git) Conda
Build type: Release
Branch: main
Hash: 173ff4da1ef4f075d56c0dc617d1bf00d5f49d4b
Python 3.11.11, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
Installed mods: 
  * Beltrami 1.1.0
  * CommandPanel
  * CurvedShapes 1.0.13
  * Curves 0.6.52
  * dodo 1.0.1
  * ExplodedAssembly
  * fasteners 0.5.33
  * frame 0.1.1
  * FrameForge 0.1.0
  * FreeCAD-Ribbon 1.4.16
  * freecad.gears 1.3.0
  * FreeGrid 2.1.1
  * OpenTheme 2024.9.1
  * SearchBar 1.2.1
  * sheetmetal 0.6.11
  * Silk 0.1.5
  * symbols_library

**Comment by APEbbers** [comment]
Hi,

Can you test if the issue is fixed in the develop branch? I'm doing the last test before the release of version 1.5.0

**Comment by raindropsfromsky** [comment]
I have already enabled the development branch
![image](https://github.com/user-attachments/assets/fb2f64d0-2c25-4949-a46c-058dc7df36bf)

I am using development branch 1.4.16 of Ribbon WB.
My git is unable to pick up 1.5.0.
I manually checked GitHub, but it also shows 1.4.16 as the latest version in the development branch.
Am I missing something?

**Comment by raindropsfromsky** [comment]
I switched to the pre-relese 1.5.0 branch, but there is no asset provided there.
It still shows 1.4.16 as the asset.

![image](https://github.com/user-attachments/assets/c5c222bd-7021-4deb-80f3-bcca7d2ed706)

**Comment by APEbbers** [comment]
Ah yes...,

You have to enable "Show option to change branches"
![image](https://github.com/user-attachments/assets/c15bc71e-5d3f-4277-9b8d-428abbae69a0)
  
Then in the addon manager you can switch branches:
![image](https://github.com/user-attachments/assets/b8fad883-665b-4216-a069-8e066ad3fc4a)

**Comment by APEbbers** [comment]
> I switched to the pre-relese 1.5.0 branch, but there is no asset provided there. It still shows 1.4.16 as the asset.
> 
> ![image](https://private-user-images.githubusercontent.com/9047168/398151547-c5c222bd-7021-4deb-80f3-bcca7d2ed706.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ5NTQ4MzEsIm5iZiI6MTczNDk1NDUzMSwicGF0aCI6Ii85MDQ3MTY4LzM5ODE1MTU0Ny1jNWMyMjJiZC03MDIxLTRkZWItODBmMy1iY2NhN2QyZWQ3MDYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMjNUMTE0ODUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDhlNGZkNzEyYzgxNjBkNjRlMWU3MDE4YWI0N2EwNWI1MmRlNzhjMTMzNjEzMDRjZDFlODAyNmViNGUwNGQ3YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.rQLFRqCc0AvkzP56y9hyk206qjzki_xqWI9wwavb8yk)

You have to xlixk on "+7 releases"

**Comment by raindropsfromsky** [comment]
I installed 1.5.0 and restarted FC.

When I launch a sketch, FC switches to  Sketcher WB automatically (as expected).
But the ribbon is completely blank! **[BUG]**
![image](https://github.com/user-attachments/assets/15ee9d4e-6bfe-4ee1-9f14-fa7e4c57112b)

If I switch to some other WB and switch back to Sketcher WB, all buttons reappear.

**Comment by raindropsfromsky** [comment]
While installing 1.5.0, I faced some error messages about uncommitted changes.
I ignored them and installed the WB.
Should I report them here?

**Comment by raindropsfromsky** [comment]
While trying to troubleshoot, I found that the Addon Manager still shows 1.4.16 as the version installed, and "up-to-date" status.
![image](https://github.com/user-attachments/assets/9d67dd17-c418-4c69-b17a-5fc2265241a0)

But the **Help** menu of FC shows version 1.5.0
![image](https://github.com/user-attachments/assets/881a8dd3-516a-490d-a45c-3fe516e6a0ad)

I hope the installation is not messed up!

**Comment by APEbbers** [comment]
No, that's probally because the main branch is still on v1.416. If the help menu show v1.5.0, your on the develop branch or pre-release branch.

**Comment by APEbbers** [comment]
> While installing 1.5.0, I faced some error messages about uncommitted changes. I ignored them and installed the WB. Should I report them here?

Did you change something in the code your self? anyway, I've updated the pre-release v1.5.0 and the develop branch.
Perhaps if you update again, this message is gone. (After update you have to switch to the develop branch again.)

**Comment by APEbbers** [comment]
> I installed 1.5.0 and restarted FC.
> 
> When I launch a sketch, FC switches to Sketcher WB automatically (as expected). But the ribbon is completely blank! **[BUG]** ![image](https://private-user-images.githubusercontent.com/9047168/398153400-15ee9d4e-6bfe-4ee1-9f14-fa7e4c57112b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ5NTU3NzgsIm5iZiI6MTczNDk1NTQ3OCwicGF0aCI6Ii85MDQ3MTY4LzM5ODE1MzQwMC0xNWVlOWQ0ZS02YmZlLTRlZTEtOWYxNC1mYTdlNGM1NzExMmIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMjNUMTIwNDM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZThjNDBmZThmZTE3ZWRlYmE4ZDFiODc4NWQwOTZmMjg2ZjkwZjA2NTlkMWM3YWJkNGFkNWVjYWFlZjllNTI3OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.NBXzOZBaIUio15ezC0hGJ8JoY3qTPlDuJPBUhx3ayr4)
> 
> If I switch to some other WB and switch back to Sketcher WB, all buttons reappear.

I can't reproduce the error. However can you open the layout menu? The first time a datafile will be generated which stores all the icons. The second time this datafile is used by the layout menu and the ribbon.

**Comment by raindropsfromsky** [comment]
I have set auto-update in Addon Manager.
So when I launched FC, it detected a new version of 1.5.0, and prompted me to update the WB.
After that, I am not facing any issues listed above.

Tried this multiple times, and found no issues...

**Comment by raindropsfromsky** [comment]
> However can you open the layout menu? The first time a datafile will be generated which stores all the icons. The second time this datafile is used by the layout menu and the ribbon.

Where is this **Layout** menu? I don't see it in the **main menu** system or in the **Preferences**.

**Comment by APEbbers** [comment]
I changed the name the name of the design menu:
![image](https://github.com/user-attachments/assets/011a3bed-c494-4ded-9ad6-1668c42cd8a4)

**Comment by raindropsfromsky** [comment]
I updated Ribbon WB through Addon Manager, and tried it out.
When I launch a new sketch, FC switches to Sketcher WB, but its ribbon is blank (as I reported before).
Switching to another WB and back restores the ribbon.

I tried it once more. 
But I simply waited this time. 
After a few seconds, the toolbars appeared in the Sketcher WB.

I tried this again for two more times, but both times the ribbon did not appear for the Sketcher WB.
I had to switch to another WB and return to see the toolbar for the Sketcher WB.

To conclude, there is a problem.

**Comment by APEbbers** [comment]
Hi,

I got finally also a blank tab for the sketcher wb. And also found a solution. With version 1.5.0.3 it should be fixed

**Comment by raindropsfromsky** [comment]
Confirmed that Sketcher ribbon pops up immediately.

Thanks!

**Comment by raindropsfromsky** [comment]
After reinstalling FC, I am getting the spurious toolbar once again!

![Image](https://github.com/user-attachments/assets/f541855b-3981-4a00-b8c4-d902865b2b5d)

OS: Windows 11 build 26100
Architecture: x86_64
Version: 1.1.0dev.40122 (Git) Conda
Build type: Release
Branch: main
Hash: c1dddaeadc68cc9f05f1312dc1c6789820cfdd5d
Python 3.11.11, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, IfcOpenShell 0.0.0, OCC 7.8.1
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: OpenDark.qss/OpenDark/Fusion
Logical/physical DPI: 96/123.928
Installed mods: 
  * Beltrami 1.1.0
  * CommandPanel
  * CurvedShapes 1.0.13
  * Curves 0.6.55
  * ExplodedAssembly
  * fasteners 0.5.37
  * frame 0.1.1
  * FrameForge 0.1.2
  * FreeCAD-Ribbon 1.8.0
  * freecad.gears 1.3.0
  * FreeGrid 2.2.0
  * OpenTheme 2024.9.1
  * parts_library
  * Quetzal 1.1.0
  * SearchBar 1.3.3
  * sheetmetal 0.7.5
  * Silk 0.1.5
  * symbols_library

**Comment by APEbbers** [comment]
There remain when you click on a tab?

**Comment by raindropsfromsky** [comment]
You mean when I change the WB? Yes.

1. The spurious toolbar is ALWAYS Sketcher toolbar (no other WB).
2. It pops up exactly at the moment when I have launched Sketcher from Parts Design WB.
    (When I select the plane on which the sketch has to be created.)
3. It vanishes when I close the Sketch and return to PD WB.
4. It does not happen every time.
